module/synchronization/metrics.go

```diff
@@ -0,0 +1,138 @@
+package synchronization
+
+import (
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type SynchronizationMetrics interface {
+	// record pruned blocks. requested and received times might be zero values
+	PrunedBlockById(status *Status)
+
+	PrunedBlockByHeight(status *Status)
+
+	// totalByHeight and totalById are the number of blocks pruned for blocks requested by height and by id
+	// storedByHeight and storedById are the number of blocks still stored by height and id
+	PrunedBlocks(totalByHeight, totalById, storedByHeight, storedById int)
+
+	RangeRequested(ran flow.Range)
+
+	BatchRequested(batch flow.Batch)
+}
+
+type NoopMetrics struct{}
+
+func (nc *NoopMetrics) PrunedBlockById(status *Status)                                        {}
+func (nc *NoopMetrics) PrunedBlockByHeight(status *Status)                                    {}
+func (nc *NoopMetrics) PrunedBlocks(totalByHeight, totalById, storedByHeight, storedById int) {}
+func (nc *NoopMetrics) RangeRequested(ran flow.Range)                                         {}
+func (nc *NoopMetrics) BatchRequested(batch flow.Batch)                                       {}
+
+const (
+	namespaceSynchronization = "synchronization"
+	subsystemSyncCore        = "sync_core"
+)
+
+type MetricsCollector struct {
+	timeToPruned          *prometheus.HistogramVec
+	timeToReceived        *prometheus.HistogramVec
+	totalPruned           *prometheus.CounterVec
+	storedByHeight        prometheus.Gauge
+	storedById            prometheus.Gauge
+	totalHeightsRequested prometheus.Counter
+	totalIdsRequested     prometheus.Counter
+}
+
+func NewMetricsCollector() *MetricsCollector {
+	return &MetricsCollector{
+		timeToPruned: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name:      "time_to_pruned",
+			Namespace: namespaceSynchronization,
+			Subsystem: subsystemSyncCore,
+			Help:      "the time between queueing and pruning a block in milliseconds",
+			Buckets:   []float64{100, 250, 500, 1000, 2500, 5000, 7500, 10000, 20000},
+		}, []string{"status", "requested_by"}),
```

Comment:
There are some conventions around Prometheus metric units and naming we try to follow which we can apply here:

>  metric names should use base units (e.g. seconds, bytes, meters - not milliseconds, megabytes, kilometers)

> should have a suffix describing the unit, in plural form. Note that an accumulating count has total as a suffix, in addition to the unit if applicable.

The full doc is [here](https://prometheus.io/docs/practices/naming/)

Example of updating to the unit convention:
```suggestion
		timeToPruned: prometheus.NewHistogramVec(prometheus.HistogramOpts{
			Name:      "time_to_pruned_seconds",
			Namespace: namespaceSynchronization,
			Subsystem: subsystemSyncCore,
			Help:      "the time between queueing and pruning a block in seconds",
			Buckets:   []float64{.1, .25, .5, 1, 2.5, 5, 7.5, 10, 20},
		}, []string{"status", "requested_by"}),
```
(can apply to the other metrics as well)